src/main/java/com/treasure/hunt/game/GameManager.java

```diff
@@ -1,26 +1,73 @@
 package com.treasure.hunt.game;
 
-import com.treasure.hunt.strategy.hint.AbstractHintGenerator;
-import com.treasure.hunt.strategy.moves.AbstractMovesGenerator;
-import com.treasure.hunt.ui.in_game.UiRenderer;
+import com.treasure.hunt.strategy.hint.Hint;
+import com.treasure.hunt.strategy.tipster.Tipster;
+import com.treasure.hunt.strategy.seeker.Moves;
+import com.treasure.hunt.strategy.seeker.Seeker;
+import com.treasure.hunt.ui.in_game.View;
 import lombok.RequiredArgsConstructor;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.GeometryFactory;
 import org.locationtech.jts.geom.Point;
 
 @RequiredArgsConstructor
 public class GameManager {
-    private final AbstractHintGenerator hintGenerator;
-    private final AbstractMovesGenerator searchStrategy;
-    private final UiRenderer uiRenderer;
-    private final GameHistory gameHistory = new GameHistory();
+    // Players
+    private final Tipster tipster;
+    private final Seeker seeker;
 
-    private Point player;
-    private Point target;
+    // Render variables
+    private final View view;
+    private final GameHistory gameHistory;
 
-    public void onNext() {
+    // Geometries
+    private GeometryFactory gf = new GeometryFactory();
+    private Point seekerPos;
+    private Point treasurePos;
+
+    // Game variables
+    private Hint lastHint;
+
+    public GameManager(Seeker seeker, Tipster tipster, View view) {
```

Comment:
this is not needed it is auto generated by  @RequiredArgsConstructor  it creates a constructor with all final variables